core/src/main/java/org/elasticsearch/search/fetch/subphase/highlight/FastVectorHighlighter.java

```diff
@@ -212,24 +219,31 @@ public boolean canHighlight(FieldMapper fieldMapper) {
 
     private static BoundaryScanner getBoundaryScanner(Field field) {
         final FieldOptions fieldOptions = field.fieldOptions();
-        final Locale boundaryScannerLocale = fieldOptions.boundaryScannerLocale();
-        switch(fieldOptions.boundaryScannerType()) {
-        case SENTENCE:
-            if (boundaryScannerLocale != null) {
-                return new BreakIteratorBoundaryScanner(BreakIterator.getSentenceInstance(boundaryScannerLocale));
-            }
-            return DEFAULT_SENTENCE_BOUNDARY_SCANNER;
-        case WORD:
-            if (boundaryScannerLocale != null) {
-                return new BreakIteratorBoundaryScanner(BreakIterator.getWordInstance(boundaryScannerLocale));
-            }
-            return DEFAULT_WORD_BOUNDARY_SCANNER;
-        default:
-            if (fieldOptions.boundaryMaxScan() != SimpleBoundaryScanner.DEFAULT_MAX_SCAN
+        final Locale boundaryScannerLocale =
+            fieldOptions.boundaryScannerLocale() != null ? fieldOptions.boundaryScannerLocale() :
+                Locale.ROOT;
+        final HighlightBuilder.BoundaryScannerType type =
```

Comment:
If I reading this right, this is the only change to the file. Everything else is just word wrapping?